ecl/eclcc/eclcc.cpp

```diff
@@ -1872,6 +1897,145 @@ bool EclCompileInstance::reportErrorSummary()
 void EclCC::noteCluster(const char *clusterName)
 {
 }
+
+IHqlExpression *EclCC::lookupDFSlayout(const char *filename, IErrorReceiver &errs, const ECLlocation &location, bool isOpt) const
+{
+    CriticalBlock b(dfsCrit);  // Overkill at present but maybe one day codegen will start threading?
+    if (!optDFS || disconnectReported)
+    {
+        // Dali lookup disabled, yet translation requested. Should we report if OPT set?
+        if (!(optArchive || optGenerateDepend || optSyntax || optGenerateMeta || optEvaluateResult || disconnectReported))
+        {
+            VStringBuffer msg("Error looking up file %s in DFS - DFS not configured", filename);
+            errs.reportWarning(CategoryDFS, HQLWRN_DFSlookupFailure, msg.str(), str(location.sourcePath), location.lineno, location.column, location.position);
+            disconnectReported = true;
+        }
+        return nullptr;
+    }
+    if (!daliConnected)
+    {
+        try
+        {
+            Owned<IGroup> serverGroup = createIGroup(optDFS.str(), DALI_SERVER_PORT);
+            if (!initClientProcess(serverGroup, DCR_EclCC, 0, NULL, NULL, optDaliTimeout))
+            {
+                VStringBuffer msg("Error looking up file %s in DFS - failed to connect to %s", filename, optDFS.str());
+                errs.reportError(HQLWRN_DFSlookupFailure, msg.str(), str(location.sourcePath), location.lineno, location.column, location.position);
+                disconnectReported = true;
+                return nullptr;
+            }
+        }
+        catch (IException *E)
+        {
+            StringBuffer emsg;
+            VStringBuffer msg("Error looking up file %s in DFS - failed to connect to %s (%s)", filename, optDFS.str(), E->errorMessage(emsg).str());
+            E->Release();
+            errs.reportError(HQLWRN_DFSlookupFailure, msg.str(), str(location.sourcePath), location.lineno, location.column, location.position);
+            disconnectReported = true;
+            return nullptr;
+
+        }
+        daliConnected = true;
+    }
+
+    // Do any scope manipulation
+    StringBuffer lookupName;  // do NOT move inside the curly braces below - this needs to stay in scope longer than that
+    if (filename[0]=='~')
+        filename++;
+    else if (!optScope.isEmpty())
+    {
+         lookupName.appendf("%s::%s", optScope.str(), filename);
+         filename = lookupName.str();
+    }
+
+    // First lookup the name in our cache...
+    Linked<IHqlExpression> ret = fileCache.getValue(filename);
+    if (ret)
+        return ret.getClear();
+
+    int err = 0;
+    OwnedHqlExpr diskRecord;
+
+    // check the nohit cache...
+    int *nohit = fileMissCache.getValue(filename);
+    if (nohit)
+        err = *nohit;
+    else
+    {
+        // Look up the file in Dali
+        Owned<IUserDescriptor> udesc;
```

Comment:
Probably not, but logically probably makes sense to do so since the cache is per user.